### PR TITLE
Refactor routes & helpers

### PR DESCRIPTION
**Pull request by florent-uzio**
This PR refactors the Routes to make it more flexible and readable.

It also adds some helpers to:
- Easily inject a path parameter in a URL (+ tests)
- Identify if a const is a string, number, object, function... (+ tests)

GemWallet still works as usual:

https://github.com/GemWallet/gemwallet-extension/assets/36513774/666b5b76-dc6c-4567-84ce-7def4a0202a2

### CODE REVIEW

**Comment by FlorianBouron** [comment]
@all-contributors add @florent-uzio for code

File: packages/extension/src/constants/paths.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const CANCEL_NFT_OFFER_PATH = '/cancel-nft-offer';
 export const CANCEL_OFFER_PATH = '/cancel-offer';
 export const CREATE_NFT_OFFER_PATH = '/create-nft-offer';
 export const CREATE_OFFER_PATH = '/create-offer';
-export const EDIT_WALLET_PATH = '/edit-wallet';
+export const EDIT_WALLET_PATH = '/edit-wallet/:publicAddress';
```
Comment (FlorianBouron) [inline packages/extension/src/constants/paths.ts:12]: Nice catch, thanks 👍

**Comment by FlorianBouron** [comment]
Thank you so much for your contribution, it seems ready to merge.
Though could you please remove the unused constant within `App.tsx`, here is the link to the failing pipeline: https://github.com/GemWallet/gemwallet-extension/actions/runs/5835263193/job/15827483774?pr=239

**Review by FlorianBouron — APPROVED** [review]
Approved with the following comment above

**Comment by ThibautBremand** [comment]
Thanks a lot @florent-uzio 👍

**Pull request merged**